src/repos/common/save-event.js

```diff
@@ -7,12 +24,17 @@ const saveEvent = async (collection, event) => {
     throw new Error('Correlation ID is required')
   }
 
+  if (await checkIdempotency(collection, event)) {
+    return logger.warn(`Idempotency check failed, event not saved ${event.id}`)
+  }
+
   await db.collection(collection).updateOne(
     { _id: correlationId },
     { $push: { events: event } },
     { upsert: true }
-
   )
+
+  return logger.info(`Event processed successfully, eventId ${event.id}`)
```

Comment:
I think the caller should handle logging of the outcome (Separation of Responsibility). Here, you have added another responsibility of logging transaction progress to your saveEvent function.